### ISSUE TITLE
Validation feature

### ISSUE THREAD

**Issue by HeinrichReimer** [HeinrichReimer/material-intro#7]
Would be nice to have the ability to block users from getting past a given page based on some sort of policy:

````java
public interface PageContinuePolicy{
    boolean canGoToPage(int oldPosition, int newPosition);
}
````

That way we could also check for permissions and such. You could also disallow users to go back.

**Comment by AkshayChordiya** [comment]
@HeinrichReimer Sure, there's a need of such policy. Hoping to see it soon. Thank you.

**Comment by Jucedik** [comment]
Hi.
Please apply my pull request.
I added possibility which allow to lock and unlock next/previous (and swipe left/right) navigation.

Instructions:
**1**. Create FragmentSlide with build option "allowNext(boolean value)" or "allowPrevious(boolean value)". Warning!!! - Your fragment should be extended from SlideFragment. E.g.:
```
slides.add(new FragmentSlide.Builder()
                    .background(R.color.color_custom_fragment_1)
                    .backgroundDark(R.color.color_dark_custom_fragment_1)
                    .fragment(LoginFragment.newInstance())
                    **.allowNext(false)**
                    .build());
```

**2**.  Set NavigationInterface in MaterialIntroActivity by calling
setNavigationInterface(NavigationInterface navigationInterface). E.g.:
 ```
setNavigationInterface(new NavigationInterface() {
            @Override
            public boolean onNextClick(int position) {
                return slides.get(position).isAllowSlideNext();
            }

            @Override
            public boolean onPreviousClick(int position) {
                return slides.get(position).isAllowSlidePrevious();
            }

            @Override
            public void onImpossibleToNavigate(int position, int direction) {
                Toast.makeText(getApplicationContext(), R.string.toast_impossible_to_navigate, Toast.LENGTH_LONG).show();
            }
        });
```

**3**. For lock or unlock slide page you need to call **allowNextSlide()** or **allowPreviousSlide()** from your fragment (e.g.: you may call this methods after successfull user autorization). Also you may call **showNextSlide()** and **showPreviousSlide()** for imidiately go to next or previous slide.

You can see the sample code in demo files.

**Comment by HeinrichReimer** [comment]
Fixed by #24
